pkg/cosign/verify.go

```diff
@@ -1281,3 +1292,57 @@ func correctAnnotations(wanted, have map[string]interface{}) bool {
 	}
 	return true
 }
+
+// verifyImageSignaturesExperimentalOCI does all the main cosign checks in a loop, returning the verified signatures.
+// If there were no valid signatures, we return an error, using OCI 1.1+ behavior.
+func verifyImageSignaturesExperimentalOCI(ctx context.Context, signedImgRef name.Reference, co *CheckOpts) (checkedSignatures []oci.Signature, bundleVerified bool, err error) {
```

Comment:
Seems to be a fair bit of duplication here.

What do you think about:

1. Keeping one `verifyImageSignatures` function
2. Factor out a `fetchImageSignatures` and `fetchImageSignaturesReferrer` and call that from inside the `if sigRef == ""` block, gated on `co.RegistryOptions.SupportReferrers` (or whatever)

?


(Genuinely asking, I could go either way)